tao_compiler/mlir/disc/transforms/lower_tf.cc

```diff
@@ -1080,6 +1083,161 @@ LogicalResult convertQuantizedConv2DWithBiasAndRequantizeOp(func::FuncOp func) {
   return success();
 }
 
+// Returns 1D 64-bit dense elements attribute with the given values.
+static mlir::DenseIntElementsAttr GetI64ElementsAttr(
+    absl::Span<const int64_t> values, Builder* builder) {
+  auto ty = mlir::RankedTensorType::get({static_cast<int64_t>(values.size())},
+                                        builder->getIntegerType(64));
+  return mlir::DenseIntElementsAttr::get(
+      ty, llvm::makeArrayRef(values.data(), values.size()));
+}
+
+// Gets all float values from the given attribute and push them to `values`.
+static DenseFPElementsAttr GetF32ElementsAttr(Attribute attr, Builder *builder) {
+  SmallVector<float, 4> values;
+  auto array_attr = attr.cast<ArrayAttr>();
```

Comment:
have a check on `array_attr`?

Reply:
According to llvm [docs](https://mlir.llvm.org/doxygen/classmlir_1_1Attribute.html#a711073f32e43429d5faaada8c387ff88), `cast` API has a `assert` check first. Thus we switch to use `dyn_cast` which will return a nullptr when type is not compatible.